src/main/java/com/shrralis/ssdemo1/service/SocialService.java

```diff
@@ -58,29 +49,37 @@ public User googleProfileExtract(Connection<Google> connection){
 		if(user == null){
 			user = new User();
 			user.setEmail(person.getAccountEmail());
-			user.setLogin("google" + RandomStringUtils.randomAlphabetic(8));
+			SocialService.findLogin(user);
 			user.setPassword(UUID.randomUUID().toString());
 			String name = person.getGivenName();
+			SocialService.validateAndSetName(user, name);
 			String surname = person.getFamilyName();
-			SocialService.validateName(user, name);
-			SocialService.validateSurname(user, surname);
+			SocialService.validateAndSetSurname(user, surname);
 		}
 		return user;
 	}
 
-	private static void validateName(User user, String name){
+	private static void validateAndSetName(User user, String name){
 		if((name.length() > MIN_NAME_LENGTH && name.length() < MAX_NAME_LENGTH) && name.matches(NAME_PATTERN)){
 			user.setName(name);
 		} else{
-			user.setName("Name");
+			user.setName(null);
 		}
 	}
 
-	private static void validateSurname(User user, String surname){
-		if((surname.length() > MIN_SURNAME_LENGTH && surname.length() < MAX_SURNAME_LENGTH) && surname.matches(NAME_PATTERN)){
+	private static void validateAndSetSurname(User user, String surname){
+		if((surname.length() > MIN_NAME_LENGTH && surname.length() < MAX_NAME_LENGTH) && surname.matches(NAME_PATTERN)){
 			user.setSurname(surname);
 		} else{
-			user.setSurname("Surname");
+			user.setSurname(null);
+		}
+	}
+
+	private static void findLogin(User user){
```

Comment:
Method name is not correct.

Reply:
Can be done in one line:
user.setLogin(StringUtils.substring(user.getLogin().replace("@", ""), 0, 16))